### PR TITLE
feat: add initial site edit navbar

### PR DESCRIPTION
**Pull request by seaerchin**
### TL;DR
This pull request refactors the schema and component structure related to the editing experience.

### What changed?
- Moved `editPageSchema` to a centralized `schema.ts` file in the `editing-experience` feature.
- Introduced `SiteEditNavbar` component to improve the navigational structure within the editing experience.
- Adjusted `PageEditingLayout` to use the new `SiteEditNavbar` component.
- Cleaned up imports and removed redundant code for better maintainability.

### How to test?
1. Navigate to any page edit view.
2. Verify that the breadcrumb navigation renders correctly and links function as expected.
3. Ensure the entire editing experience functions without errors.

### Why make this change?
To improve code organization, enhance the user interface, and make future maintenance easier.

### CODE REVIEW

**Comment by seaerchin** [comment]
* **#374** <a href="https://app.graphite.dev/github/pr/opengovsg/isomer/374?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @seaerchin and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: apps/studio/src/features/editing-experience/components/SiteEditNavbar.tsx [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+      >
+        <Breadcrumb size="xs">
+          <BreadcrumbItem>
+            <BreadcrumbLink as={Link} href={`/sites/${siteId}`}>
+              All pages
+            </BreadcrumbLink>
+          </BreadcrumbItem>
+
+          <BreadcrumbItem isCurrentPage>
+            <BreadcrumbLink href="#">Current page</BreadcrumbLink>
```
Comment (dcshzj) [inline apps/studio/src/features/editing-experience/components/SiteEditNavbar.tsx:36]: question(non-blocking): is this copy supposed to be "Current page"? I suppose we are supposed to display the title here or something?

**Review by dcshzj — APPROVED** [review]
Going to assume that more work will be done on this navbar

File: apps/studio/src/features/editing-experience/components/SiteEditNavbar.tsx (hunk 1 above)
Comment (seaerchin) [reply apps/studio/src/features/editing-experience/components/SiteEditNavbar.tsx:36]: yeap it is, checking w designers. lemme merge this PR first and do ^ in a follow up

**Pull request merged**